### PR TITLE
Fixes Issue#3493: Fixed margin for widgets with buttons when value is added manually

### LINKED ISSUE #3493: Margin is not noticeable for widgets with buttons when a value is added manually
#### Software and hardware versions 
Collect v1.25

#### Problem description

Margin is NOT visible for widgets:
* Ex string when a value is added manually
* Ex integer when a value is added manually
* Ex decimal when a value is added manually
* Bearing when a value is added manually

#### Steps to reproduce the problem
1. User opens one on mentioned widgets
2. User clicks on the button
3. Text field appears
4. Margin is not visible for the text field

#### Expected behavior
Margin is noticeable 

#### Other information 
Noticed during testing #3464

### PR DESCRIPTION
**Pull request by SaumiaSinghal**
Closes #3493 

#### What has been done to verify that this works as intended?
I tested it on Android 9.0.

Before adding margin           |     After adding margin
:-------------------------:|:-------------------------:
Screenshot_20191230_034615  |  Screenshot_20191230_033317

   

#### Why is this the best possible solution? Were any other approaches considered?
I used the default margin value, that is being used in the other widgets, for uniformity. It adds the margin to the answer layout, where the answer field will appear. 

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I don't think there are any regression risks.

#### Do we need any specific form for testing your changes? If so, please attach one.
Allwidges or any form that has any of the three widgets:

- Ex-integer widget
- Ex-decimal widget
- Ex-string widget 

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
no

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]
The issue is related to all widgets that contain buttons but you implemented changes only in ExStringWidget class, that's not enough. Please investigate [All Widgets form](https://docs.google.com/spreadsheets/d/1af_Sl8A_L8_EULbhRLHVl8OclCfco09Hq2tqb9CslwQ/edit#gid=0) to get to know more about different types of widgets and complete your solution.

**Comment by grzesiek2010** [comment]
Apart from the fact this solution is not complete @seadowg isn't it something that you are working on as a part of another issue/pr? I'm asking just to avoid conflicts.

**Comment by seadowg** [comment]
@grzesiek2010 @SaumiaSinghal I think we'd want to revise the layout of these widgets after we look at changing up the widget framework. That'll be after #3483 (which doesn't look like it conflicts with this) so I think it's ok to go ahead with an earlier fix as long as it's not introducing any large changes.

**Comment by SaumiaSinghal** [comment]
Hie @grzesiek2010, I fixed it for all the widgets, wherever I found this problem. Can you review it again?

File: collect_app/src/main/java/org/odk/collect/android/widgets/AbstractSelectOneWidget.java
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/widgets/AbstractSelectOneWidget.java]: We don't want to any of the select widgets as we need the select options to spread from edge to edge. This change should be reverted.

File: collect_app/src/main/java/org/odk/collect/android/widgets/AnnotateWidget.java
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/widgets/AnnotateWidget.java:70]: It'd be good to pull this repeated code into `WidgetViewUtils`. Could be a method like `getStandardMargin()`.

**EDIT**: `WidgetViewUtils is actually not in the code yet as it is introduced in #3483. I think we should block this PR on that one as we may end up with conflicts. It should hopefully be merged soon 🤞.

File: collect_app/src/main/java/org/odk/collect/android/widgets/ArbitraryFileWidget.java
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/widgets/ArbitraryFileWidget.java]: This `FIELD_HORIZONTAL_MARGIN_MODIFIER` should only really be used on the `StringWidget` as it will make the margin smaller than it should be. Could you make sure to only modify the standard margin on the widgets that you feel need it? If it's not just the `StringWidget` that needs it we should move the constant somewhere else.

File: collect_app/src/main/java/org/odk/collect/android/widgets/ExStringWidget.java
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/widgets/ExStringWidget.java:124]: This margin ends up being too big as `QuestionWidget#getSimpleButton` is adding a margin on for us. Your changes should allow us to delete the code that adds the margin there so you should be able to do that! Again though, I'd wait for #3483 to be merged as this method moves to `WidgetViewUtils`.

**Review by seadowg — CHANGES_REQUESTED** [review]
A few changes needed. I'd wait for #3483 to be merged before going ahead though as that will give you some help.

File: collect_app/src/main/java/org/odk/collect/android/widgets/AnnotateWidget.java
Comment (SaumiaSinghal) [reply collect_app/src/main/java/org/odk/collect/android/widgets/AnnotateWidget.java:70]: Yes, @seadowg, I also think that this bit of code is repetitive, and better should be specified in a separate class. It would be nice to have the PR merged with `WidgetViewUtils`, after that I can shift it there.

**Comment by SaumiaSinghal** [comment]
Hie @seadowg, yes it'd be better to wait for #3483 to be merged then, we can decide which all widgets need the margin and if we are extending the margin not only to the string widgets, but also to others as well, we can specify the constant `FIELD_HORIZONTAL_MARGIN_MODIFIER` somewhere else. Once, #3483 is merged, I will rework on the margin issue.

**Comment by seadowg** [comment]
@SaumiaSinghal #3483 is merged now 👍

**Comment by SaumiaSinghal** [comment]
Hie @seadowg! I shifted the reused code to `WidgetViewUtils` and added margins using  `WidgetViewUtils.getStandardMargin()` to the widgets, where I found it necessary. I'm closing this PR and opening a new one.

**Pull request closed without merging**
